### PR TITLE
fix: make sure we don't initialize mono before calling _start

### PR DESCRIPTION
**Pull request by mhmd-azeez**
Currently, we are exporting an `_initialize` function that initializes the mono runtime. This is called by the host before calling the exported functions. However, the Rust SDK seems to call `_initialize` before `_start` too, which causes a problem because `_start` initializes the mono runtime too.

To fix this, I have changed the behavior of the PDK to NOT export `_initialize` anymore. Instead, we manually initialize the mono runtime before invoking any of the exported functions. This is inline with the behavior of some of the other compilers, namely: Rust ~~and Haskell~~.

This has another added benefit: wasmtime refused to call plugins that had both `_initialize` and `_start` functions exported. This is no longer an issue since we never export `_initialize` anymore

### CODE REVIEW

File: samples/SampleHost/Program.cs [hunk 1]
```diff
@@ -14,11 +14,9 @@
 
 void IsVowel(CurrentPlugin plugin, Span<ExtismVal> inputs, Span<ExtismVal> outputs, nint userData)
 {
-    var bytes = plugin.ReadBytes(inputs[0].v.i32);
+    var c = (char)inputs[0].v.i32;
```
Comment (nilslice) [inline samples/SampleHost/Program.cs:17]: is there any need for a length check here?

Comment (mhmd-azeez) [reply samples/SampleHost/Program.cs:17]: I don't think so, the plugin won't be able to call the host function if they don't import it with the right signature

Comment (nilslice) [reply samples/SampleHost/Program.cs:17]: ok cool -- thank you!

**Review by nilslice — APPROVED** [review]
@zshipko or @bhelx would be better suited to give a final :+1: on this, but it LGTM

**Review by zshipko — APPROVED** [review]

**Pull request merged**